src/main/resources/META-INF/rewrite/geotools-34.yml

```
@@ -0,0 +1,192 @@
+      newFullyQualifiedTypeName: org.geotools.api.style.StyledLayerDescriptor
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.styling.Style
+      newFullyQualifiedTypeName: org.geotools.api.style.Style
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.styling.Displacement
+      newFullyQualifiedTypeName: org.geotools.api.style.Displacement
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.styling.AnchorPoint
+      newFullyQualifiedTypeName: org.geotools.api.style.AnchorPoint
+  - org.openrewrite.java.ChangePackage:
+      oldPackageName: org.opengis
+      newPackageName: org.geotools.api
+      recursive: true
+  - org.openrewrite.java.ChangePackage:
+      oldPackageName: org.geotools.util
+      newPackageName: org.geotools.data.util
+      recursive: true
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.data.util.WeakCollectionCleaner
+      newFullyQualifiedTypeName: org.geotools.util.WeakCollectionCleaner
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.factory.FactoryNotFoundException
+      newFullyQualifiedTypeName: org.geotools.util.factory.FactoryNotFoundException
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.data.FeatureSource
+      newFullyQualifiedTypeName: org.geotools.api.data.FeatureSource
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.data.Query
+      newFullyQualifiedTypeName: org.geotools.api.data.Query
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.data.QueryCapabilities
+      newFullyQualifiedTypeName: org.geotools.api.data.QueryCapabilities
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.data.FeatureWriter
+      newFullyQualifiedTypeName: org.geotools.api.data.FeatureWriter
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.data.Transaction
+      newFullyQualifiedTypeName: org.geotools.api.data.Transaction
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.data.DataStoreFinder
+      newFullyQualifiedTypeName: org.geotools.api.data.DataStoreFinder
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.data.DataAccess
+      newFullyQualifiedTypeName: org.geotools.api.data.DataAccess
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.data.DataStore
+      newFullyQualifiedTypeName: org.geotools.api.data.DataStore
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.data.FileDataStoreFinder
+      newFullyQualifiedTypeName: org.geotools.api.data.FileDataStoreFinder
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: org.geotools.api.data.Join
+      newFullyQualifiedTypeName: org.geotools.data.Join
```

Judging by the others I'm seeing, isn't this the wrong way round?
```suggestion
  - org.openrewrite.java.ChangeType:
      oldFullyQualifiedTypeName: org.geotools.data.Join
      newFullyQualifiedTypeName: org.geotools.api.data.Join
```